packages/pwa-kit-dev/src/utils/script-utils.test.js

```diff
@@ -0,0 +1,300 @@
+        })
+    })
+
+    describe('pushBundle', () => {
+        test.each([
+            [
+                {
+                    projectSlug: 'project-slug',
+                    targetSlug: undefined,
+                    expectedURL: 'https://cloud.mobify.com/api/projects/project-slug/builds/',
+                    status: 200
+                }
+            ],
+            [
+                {
+                    projectSlug: 'project-slug',
+                    targetSlug: 'target-slug',
+                    expectedURL:
+                        'https://cloud.mobify.com/api/projects/project-slug/builds/target-slug/',
+                    status: 200
+                }
+            ],
+            [
+                {
+                    projectSlug: 'project-slug',
+                    targetSlug: undefined,
+                    expectedURL: 'https://cloud.mobify.com/api/projects/project-slug/builds/',
+                    status: 401
+                }
+            ]
+        ])(
+            'should push a built bundle and handle status codes (%p)',
+            async ({projectSlug, targetSlug, expectedURL, status}) => {
+                const message = 'message'
+                const bundle = await upload2.createBundle({
+                    message,
+                    ssr_parameters: {},
+                    ssr_only: ['*.js'],
+                    ssr_shared: ['**/*.*'],
+                    buildDirectory: path.join(__dirname, 'test-fixtures', 'minimal-built-app'),
+                    projectSlug
+                })
+
+                const username = 'user123'
+                const api_key = '123'
+                const credentials = {username, api_key}
+
+                const reponseBodyMock = {anything: 'anything'}
+                const responseMock = {status, json: () => Promise.resolve(reponseBodyMock)}
+                const fetchMock = jest.fn(async () => responseMock)
+
+                const client = new upload2.CloudAPIClient({credentials, fetch: fetchMock})
+
+                const fn = async () => await client.push(bundle, projectSlug, targetSlug)
+
+                if (status === 200) {
+                    expect(await fn()).toBe(reponseBodyMock)
+                } else {
+                    await expect(fn).rejects.toThrow('For more information visit')
+                }
+
+                expect(fetchMock.mock.calls.length).toBe(1)
+
+                const url = fetchMock.mock.calls[0][0]
+                const opts = fetchMock.mock.calls[0][1]
+
+                expect(url).toEqual(expectedURL)
+                expect(opts.body).toEqual(expect.anything(Buffer))
+                expect(opts.method).toEqual('POST')
+                expect(opts.headers['Authorization']).toMatch(new RegExp('^Basic '))
+                expect(opts.headers['Content-Length']).toEqual(opts.body.length.toString())
+                expect(opts.headers['User-Agent']).toEqual(`${pkg.name}@${pkg.version}`)
```

Comment:
```suggestion
                expect(fetchMock).toHaveBeenCalledTimes(1)

                expect(fetchMock).toHaveBeenCalledWith(
                  expectedURL,
                  expect.objectContaining({
                    body: expect.anything(Buffer),
                    method: 'POST',
                    headers: {
                      Authorization: expect.stringMatching(/^Basic /),
                      'Content-Length': opts.body.length.toString(),
                      'User-Agent': `${pkg.name}@${pkg.version}`
                    }
                  })
```
Avoid poking jest internals. 😉

Reply:
OMG so much better. TIL `.objectContaining`!

Reply:
OMG so much better. TIL `.objectContaining`!